examples/gno.land/r/gnoland/valopers/valopers.gno

```diff
@@ -5,35 +5,154 @@ package valopers
 import (
 	"std"
 
-	"gno.land/p/demo/ownable"
-	govdao "gno.land/r/gov/dao"
+	"gno.land/p/demo/avl"
+	"gno.land/p/demo/ufmt"
 )
 
-// Valoper represents a validator operator profile.
+const daoPkgPath = "gno.land/r/gov/dao"
+
+const (
+	errValoperExists        = "valoper already exists"
+	errValoperMissing       = "valoper does not exist"
+	errInvalidAddressUpdate = "valoper updated address exists"
+
+	errNotGovDAO = "caller not govdao executor"
+)
+
+// valopers keeps track of all the active validator operators
+var valopers *avl.Tree // Address -> Valoper
+
+// Valoper represents a validator operator profile
 type Valoper struct {
-	ownable.Ownable // Embedding the Ownable type for ownership management.
+	Name        string // the display name of the valoper
+	Description string // the description of the valoper
 
-	DisplayName   string      // The display name of the valoper.
-	ValidatorAddr std.Address // The address of the validator.
-	// TODO: Add other valoper metadata as needed.
+	Address      std.Address // The bech32 gno address of the validator
+	PubKey       string      // the bech32 public key of the validator
+	P2PAddresses []string    // the publicly reachable P2P addresses of the validator
 }
 
-// Register registers a new valoper.
-// TODO: Define the parameters and implement the function.
-func Register( /* TBD */ ) {
-	panic("not implemented")
+// Register registers a new valoper
+func Register(v Valoper) {
+	assertGovDAOCaller()
+
+	// Check if the valoper is already registered
+	if isValoper(v.Address) {
+		panic(errValoperExists)
+	}
+
+	// TODO add address derivation from public key
+	// (when the laws of gno make it possible)
+
+	// Save the valoper to the set
+	valopers.Set(v.Address.String(), v)
 }
 
-// Update updates an existing valoper.
-// TODO: Define the parameters and implement the function.
-func Update( /* TBD */ ) {
-	panic("not implemented")
+// Unregister unregisters a valoper
+func Unregister(address std.Address) {
```

Comment:
I propose that we do not allow unregistering, but instead we can let a validator declare that they want to stop, like `Disable()`.

Reply:
I've added an `Active` flag to indicate if the valoper is active, and removed the `Unregister` function, since the valoper can update this field already through `Update`:

[8fc9310](https://github.com/gnolang/gno/pull/2509/commits/8fc93107cec8eb94c4947659bf94da575fdb58c7)